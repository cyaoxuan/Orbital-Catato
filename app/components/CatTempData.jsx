// Temp data to showcase static pages

const catViewProps = [{
    iconName1: "information-circle",
    field1: "Status: ",
    iconName2: "location",
    field2: "Last Seen: "
}, {
    iconName1: "time",
    field1: "Last Fed: ",
    iconName2: "location",
    field2: "Last Seen: "
}];

const concernCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Engineering"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Skitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "New",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Utown"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Mitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Missing",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Science"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Bitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Arts"
}];

const fedCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[1].iconName1,
    field1: catViewProps[1].field1,
    info1: "19/05 12:01",
    iconName2: catViewProps[1].iconName2,
    field2: catViewProps[1].field2,
    info2: "Engineering"
}];

export { concernCatCards, fedCatCards }