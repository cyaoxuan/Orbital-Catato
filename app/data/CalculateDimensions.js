import { Dimensions } from "react-native";

// Get size of preview photos
export function getItemWidth(noOfItems, margin) {
    const { height, width } = Dimensions.get("window");
    // Width of screen - margins between images (8 * margin) / 4
    const itemSize = (width - margin * 2 * noOfItems) / noOfItems;
    return itemSize;
}
