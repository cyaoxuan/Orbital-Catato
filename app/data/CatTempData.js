// Temp data to showcase static pages
const cats = [{
    catID: 1,
    name: "Kitty",
    photoURLs: [require("../../assets/temp-cat.jpg"), require("../../assets/catato-logo.png")],
    gender: "M",
    birthYear: 2019,
    sterilised: true,
    keyFeatures: "Grey tabby and loves to sleep on his back",
    lastSeenLocation: "Engineering", // Will eventually be a geohash
    lastSeenTime: new Date(2023, 4, 19, 14, 21, 0, 0), // Will be converted from firebase timestamp
    lastFedTime: new Date(2023, 4, 19, 12, 1, 0, 0), 
    concernStatus: ["Injured"], 
    concernDesc: "Seen limping, possibly right front leg injured"
}, {
    catID: 2,
    name: "Skitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: null,
    birthYear: null,
    sterilised: false,
    keyFeatures: null,
    lastSeenLocation: "Utown",
    lastSeenTime: new Date(2023, 4, 20, 10, 53, 0, 0),
    lastFedTime: null,
    concernStatus: ["New", "Injured"],
    concernDesc: "New and had wounds on body"
}, {
    catID: 3,
    name: "Mitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: "F",
    birthYear: 2020,
    sterilised: true,
    keyFeatures: "White with orange patches with a heart-shaped patch on its back, tends to climb up to high places",
    lastSeenLocation: "Science",
    lastSeenTime: new Date(2023, 4, 15, 18, 34, 0, 0),
    lastFedTime: new Date(2023, 4, 15, 18, 34, 0, 0),
    concernStatus: ["Missing"], // The database registers as missing past certain time?
    concernDesc: "Has not been seen in 5 days"
}, {
    catID: 4,
    name: "Bitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: "M",
    birthYear: 2022,
    sterilised: true,
    keyFeatures: "Slightly chubby tabby and meows alot",
    lastSeenLocation: "Arts",
    lastSeenTime: new Date(2023, 4, 20, 9, 16, 0, 0),
    lastFedTime: new Date(2023, 4, 19, 20, 47, 0, 0),
    concernStatus: [], // If the array is empty it's healthy
    concernDesc: null
}];

export { cats }