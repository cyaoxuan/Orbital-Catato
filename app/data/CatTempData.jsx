// Temp data to showcase static pages
const cats = [{
    // For Select FlatList
    catID: 0,
    name: "New Cat",
    photoURL: require("../../assets/placeholder.png"),
    gender: null,
    birthYear: null,
    sterilised: null,
    keyFeatures: null,
    lastSeenLocation: null,
    lastSeenTime: null,
    lastFedTime: null,
    concernStatus: null,
    concernDesc: null
},  {
    catID: 1,
    name: "Kitty",
    photoURL: require("../../assets/temp-cat.jpg"), // Will eventually be an array of 
    gender: "M",
    birthYear: 2019,
    sterilised: true,
    keyFeatures: "Grey tabby and loves to sleep on his back",
    lastSeenLocation: "Engineering", // Will eventually be a geohash
    lastSeenTime: new Date(2023, 5, 19, 14, 21, 0, 0), // Will be converted from firebase timestamp
    lastFedTime: new Date(2023, 5, 19, 12, 1, 0, 0), 
    concernStatus: ["Injured"], 
    concernDesc: "Seen limping, possibly right front leg injured"
}, {
    catID: 2,
    name: "Skitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: null,
    birthYear: null,
    sterilised: false,
    keyFeatures: null,
    lastSeenLocation: "Utown",
    lastSeenTime: new Date(2023, 5, 20, 10, 53, 0, 0),
    lastFedTime: null,
    concernStatus: ["New", "Injured"],
    concernDesc: "New and had wounds on body"
}, {
    catID: 3,
    name: "Mitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: "F",
    birthYear: 2020,
    sterilised: true,
    keyFeatures: "White with orange patches with a heart-shaped patch on its back, tends to climb up to high places",
    lastSeenLocation: "Science",
    lastSeenTime: new Date(2023, 5, 15, 18, 34, 0, 0),
    lastFedTime: new Date(2023, 5, 15, 18, 34, 0, 0),
    concernStatus: ["Missing"], // The database registers as missing past certain time?
    concernDesc: "Has not been seen in 5 days"
}, {
    catID: 4,
    name: "Bitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: "M",
    birthYear: 2022,
    sterilised: true,
    keyFeatures: "Slightly chubby tabby and meows alot",
    lastSeenLocation: "Arts",
    lastSeenTime: new Date(2023, 5, 20, 9, 16, 0, 0),
    lastFedTime: new Date(2023, 5, 19, 20, 47, 0, 0),
    concernStatus: [], // If the array is empty it's healthy
    concernDesc: null
}];

export { cats }