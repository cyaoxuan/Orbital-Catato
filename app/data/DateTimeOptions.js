export const dateTimeOptions = {
    timeZone: "Asia/Singapore",
    month: "2-digit",
    day: "2-digit",
    hour: "2-digit",
    minute: "2-digit",
    hour12: false
}