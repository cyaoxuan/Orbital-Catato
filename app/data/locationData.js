// JSON file format
// {
//     "User Location": {
//         "key": "1",
//         "value": "Use Current Location"
//     },
//     "The Deck": {
//         "key": "2",
//         "value": "The Deck",
//         "location": {"latitude": 1.2945075717691477, "longitude": 103.77256063897944},
//         "zone": "Arts"
//     },...
// }

const locationjson = require("./location.json");
const locations = Object.values(locationjson);
const zonejson = require("./zone.json");

export { locations, locationjson, zonejson };
