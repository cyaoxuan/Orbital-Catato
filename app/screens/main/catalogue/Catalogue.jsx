import { FlatList, View } from "react-native";
import { Text } from "react-native-paper";
import { useNavigation } from "expo-router";
import { cats } from "../../../data/CatTempData";
import { TouchableCatAvatar } from "../../../components/CatAvatar";

// Eventual Call from DB
function getCats() {
    return cats;
}

export default function Catalogue() {
    const navigation = useNavigation();

    return (
        <FlatList testID="catalogue"
            style={{ alignContent: "center" }}
            ItemSeparatorComponent={() => <View style={{height: 20}} />}
            ListHeaderComponent={ <Text variant="headlineLarge" style={{ textAlign:"center", margin: 8 }}>Meet the Cats!</Text>}
            
            data={getCats()}
            renderItem={({item}) => {
                return (
                    <TouchableCatAvatar
                        size={200}
                        photoURL={item.photoURLs[0]}
                        variant="headlineLarge"
                        name={item.name}
                        // 
                        onPress={() => {
                            navigation.navigate("CatProfile", { catID: item.catID })
                        }}
                    />
                )
            }}
        />
    );
}