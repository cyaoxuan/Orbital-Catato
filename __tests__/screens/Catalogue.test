import { cleanup, render, fireEvent } from "@testing-library/react-native";
import Catalogue from "../../app/screens/main/catalogue/Catalogue";
import { cats } from "../../app/data/CatTempData";

afterEach(cleanup);

const mockNavigate = jest.fn();

jest.mock("expo-router", () => ({
    useNavigation: () => ({
        navigate: mockNavigate,
    }),
}));

describe("<Catalogue />", () => {
    it("renders catalogue successfully", () => {
        const { getByTestId } = render(<Catalogue />);
        const carousel = getByTestId("catalogue");
        expect(carousel).toBeDefined();
    });

    it("renders catalogue with cat cards correctly", () => {
        const tree = render(<Catalogue />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("navigates to correct cat profile when card is pressed", () => {
        const { getAllByTestId } = render(<Catalogue />);

        const catAvatars = getAllByTestId("cat-avatar");
        expect(catAvatars).toHaveLength(5);

        fireEvent.press(catAvatars[0]);

        expect(mockNavigate).toHaveBeenCalledWith("CatProfile", {
            catID: cats[0].catID,
        });
    });
});
