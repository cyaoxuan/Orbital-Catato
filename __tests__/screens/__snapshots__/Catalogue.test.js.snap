// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Catalogue /> renders catalogue with cat cards correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListHeaderComponent={
    <ForwardRef(Text)
      style={
        {
          "margin": 8,
          "textAlign": "center",
        }
      }
      variant="headlineLarge"
    >
      Meet the Cats!
    </ForwardRef(Text)>
  }
  data={
    [
      {
        "birthYear": 2019,
        "catID": 1,
        "concernDesc": "Seen limping, possibly right front leg injured",
        "concernStatus": [
          "Injured",
        ],
        "gender": "M",
        "keyFeatures": "Grey tabby and loves to sleep on his back",
        "lastFedTime": 2023-05-19T12:01:00.000Z,
        "lastSeenLocation": "Engineering",
        "lastSeenTime": 2023-05-19T14:21:00.000Z,
        "name": "Kitty",
        "photoURLs": [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "sterilised": true,
      },
      {
        "birthYear": null,
        "catID": 2,
        "concernDesc": "New and had wounds on body",
        "concernStatus": [
          "New",
          "Injured",
        ],
        "gender": null,
        "keyFeatures": null,
        "lastFedTime": null,
        "lastSeenLocation": "Utown",
        "lastSeenTime": 2023-05-20T10:53:00.000Z,
        "name": "Skitty",
        "photoURLs": [
          1,
          1,
          1,
          1,
          1,
        ],
        "sterilised": false,
      },
      {
        "birthYear": 2020,
        "catID": 3,
        "concernDesc": "Has not been seen in 5 days",
        "concernStatus": [
          "Missing",
        ],
        "gender": "F",
        "keyFeatures": "White with orange patches with a heart-shaped patch on its back, tends to climb up to high places",
        "lastFedTime": 2023-05-15T18:34:00.000Z,
        "lastSeenLocation": "Science",
        "lastSeenTime": 2023-05-15T18:34:00.000Z,
        "name": "Mitty",
        "photoURLs": [
          1,
          1,
          1,
          1,
        ],
        "sterilised": true,
      },
      {
        "birthYear": 2022,
        "catID": 4,
        "concernDesc": null,
        "concernStatus": [],
        "gender": "M",
        "keyFeatures": "Slightly chubby tabby and meows alot",
        "lastFedTime": 2023-05-19T20:47:00.000Z,
        "lastSeenLocation": "Arts",
        "lastSeenTime": 2023-05-20T09:16:00.000Z,
        "name": "Bitty",
        "photoURLs": [
          1,
          1,
          1,
        ],
        "sterilised": true,
      },
      {
        "birthYear": null,
        "catID": 5,
        "concernDesc": null,
        "concernStatus": [],
        "gender": null,
        "keyFeatures": null,
        "lastFedTime": null,
        "lastSeenLocation": null,
        "lastSeenTime": null,
        "name": "Nully",
        "photoURLs": [],
        "sterilised": false,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    {
      "alignContent": "center",
    }
  }
  testID="catalogue"
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Text
        style={
          [
            {
              "fontFamily": "System",
              "fontSize": 32,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 40,
            },
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            {
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Meet the Cats!
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "opacity": 1,
          }
        }
        testID="cat-avatar"
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgba(103, 80, 164, 1)",
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              },
              undefined,
            ]
          }
        >
          <Image
            accessibilityIgnoresInvertColors={true}
            source={1}
            style={
              {
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              }
            }
            testID="avatar-image"
          />
        </View>
        <Text
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 32,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 40,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Kitty
        </Text>
      </View>
      <View
        style={
          {
            "height": 20,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "opacity": 1,
          }
        }
        testID="cat-avatar"
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgba(103, 80, 164, 1)",
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              },
              undefined,
            ]
          }
        >
          <Image
            accessibilityIgnoresInvertColors={true}
            source={1}
            style={
              {
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              }
            }
            testID="avatar-image"
          />
        </View>
        <Text
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 32,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 40,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Skitty
        </Text>
      </View>
      <View
        style={
          {
            "height": 20,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "opacity": 1,
          }
        }
        testID="cat-avatar"
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgba(103, 80, 164, 1)",
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              },
              undefined,
            ]
          }
        >
          <Image
            accessibilityIgnoresInvertColors={true}
            source={1}
            style={
              {
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              }
            }
            testID="avatar-image"
          />
        </View>
        <Text
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 32,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 40,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Mitty
        </Text>
      </View>
      <View
        style={
          {
            "height": 20,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "opacity": 1,
          }
        }
        testID="cat-avatar"
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgba(103, 80, 164, 1)",
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              },
              undefined,
            ]
          }
        >
          <Image
            accessibilityIgnoresInvertColors={true}
            source={1}
            style={
              {
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              }
            }
            testID="avatar-image"
          />
        </View>
        <Text
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 32,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 40,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Bitty
        </Text>
      </View>
      <View
        style={
          {
            "height": 20,
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "opacity": 1,
          }
        }
        testID="cat-avatar"
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgba(103, 80, 164, 1)",
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              },
              undefined,
            ]
          }
        >
          <Image
            accessibilityIgnoresInvertColors={true}
            source={1}
            style={
              {
                "borderRadius": 100,
                "height": 200,
                "width": 200,
              }
            }
            testID="avatar-image"
          />
        </View>
        <Text
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 32,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 40,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Nully
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
